apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -0,0 +1,161 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var ManifestHelper = require('shared/manifest_helper');
+  var AddonManager = require('modules/addon_manager');
+  var Toaster = require('shared/toaster');
+
+  const PREFERRED_ICON_SIZE = 64 * (window.devicePixelRatio || 1);
+
+  /**
+   * @class AddonDetails
+   * @requires module:modules/settings_service
+   * @requires module:shared/manifest_helper
+   * @requires module:modules/addon_manager
+   * @requires module:shared/toaster
+   * @returns {AddonDetails}
+   */
+  function AddonDetails(panel) {
+    this.panel = panel;
+    this.body = panel.querySelector('.addon-details-body');
+    this.header = panel.querySelector('.addon-details-header');
+    this.icon = panel.querySelector('.addon-details-icon');
+    this.description = panel.querySelector('.addon-description-text');
+    this.developer = panel.querySelector('.addon-developer');
+    this.targetsList = panel.querySelector('.addon-targets');
+    this.noTargetsMsg = panel.querySelector('.addon-no-targets');
+    this.toggle = panel.querySelector('.addon-enabled');
+    this.deleteButton = panel.querySelector('.addon-delete');
```

Comment:
Take `this.developer` for example, we can't easily recognize its use at the first glimpse.